apm-agent-plugins/apm-spring-webmvc-plugin/src/main/java/co/elastic/apm/agent/spring/webmvc/ExceptionHandlerInstrumentation.java

```diff
@@ -0,0 +1,86 @@
+/*-
+ * #%L
+ * Elastic APM Java agent
+ * %%
+ * Copyright (C) 2018 - 2019 Elastic and contributors
+ * %%
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * #L%
+ */
+package co.elastic.apm.agent.spring.webmvc;
+
+import co.elastic.apm.agent.bci.ElasticApmInstrumentation;
+import co.elastic.apm.agent.impl.ElasticApmTracer;
+import co.elastic.apm.agent.impl.stacktrace.StacktraceConfiguration;
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.description.method.MethodDescription;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+
+import javax.servlet.http.HttpServletRequest;
+import java.util.Arrays;
+import java.util.Collection;
+
+import static net.bytebuddy.matcher.ElementMatchers.named;
+import static net.bytebuddy.matcher.ElementMatchers.returns;
+import static net.bytebuddy.matcher.ElementMatchers.takesArgument;
+
+public class ExceptionHandlerInstrumentation extends ElasticApmInstrumentation {
+
+    private final Collection<String> applicationPackages;
+
+    public ExceptionHandlerInstrumentation(ElasticApmTracer tracer) {
+        applicationPackages = tracer.getConfig(StacktraceConfiguration.class).getApplicationPackages();
+    }
+
+    @Override
+    public Class<?> getAdviceClass() {
+        return ExceptionHandlerAdviceService.class;
+    }
+
+    public static class ExceptionHandlerAdviceService {
+
+        @Advice.OnMethodEnter(suppress = Throwable.class)
+        public static void captureException(@Advice.Argument(0) HttpServletRequest request,
+                                            @Advice.Argument(4) Exception e) {
```

Comment:
```suggestion
                                            @Advice.Argument(3) Exception e) {
```
This doesn't work, which means it is not tested I guess...

Reply:
Yep, tests seem to pass without this instrumentation so we currently don't have a test for it.
@kananindzya Did you change the test as well since I reviewed last?

Reply:
yes, strange behavior, the test was successful.
I changed the test. It takes time to fix it.